### PR TITLE
Add RenderViewport Command

### PR DESCRIPTION
**Pull request by jaydenmilne**
Part of BYU-PCCL/holodeck#135

Add a command to toggle rendering the viewport

### CODE REVIEW

File: Source/Holodeck/ClientCommands/Private/CommandFactory.cpp [hunk 1]
```diff
@@ -10,7 +10,8 @@ UCommand* UCommandFactory::MakeCommand(const std::string& Name, const std::vecto
 										  { "DayTime", &CreateInstance<UDayTimeCommand> },
 										  { "SetWeather", &CreateInstance<USetWeatherCommand> },
 										  { "DayCycle", &CreateInstance<UDayCycleCommand> },
-										  { "TeleportCamera", &CreateInstance<UTeleportCameraCommand> } };
+										  { "TeleportCamera", &CreateInstance<UTeleportCameraCommand> },
+										  {	"RenderViewport", &CreateInstance<URenderViewportCommand> } };
```
Comment (nickwalton) [inline Source/Holodeck/ClientCommands/Private/CommandFactory.cpp]: Nits: Just a small thing but align the renderviewport line with the others

File: Source/Holodeck/ClientCommands/Private/RenderViewportCommand.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+	const FString command = TEXT("ShowFlag.Rendering ");
+	const FString paramater = (NumberParams[0] == 1) ? TEXT("2") : TEXT("0");
+	const FString complete_command = command + paramater;
+
+	succeeded = GEngine->Exec(GetWorld(), *complete_command);
+
+	if (succeeded) {
+		UE_LOG(LogHolodeck, Log, TEXT("URenderViewportCommand was successful"));
+	}
```
Comment (nickwalton) [inline Source/Holodeck/ClientCommands/Private/RenderViewportCommand.cpp]: Another nits: The current style guide has the last brace of the if on the same line as the else. See here: https://github.com/BYU-PCCL/holodeck-engine/wiki/Style-Guide#code-blocks

File: Source/Holodeck/ClientCommands/Private/RenderViewportCommand.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	if (StringParams.size() != 0 || NumberParams.size() != 1) {
+		UE_LOG(LogHolodeck, Error, TEXT("Unexpected argument length found in URenderViewportCommand. Command not executed."));
+		return;
+	}
+
+	bool succeeded;
+
+	// TODO: Apparently you aren't "supposed" to just manually send a command like this.
+	//       It would be better to get a reference to the ViewFamily to do `ViewFamily.EngineShowFlags.Rendering = 0`;
+	//		 This is basically what the command below does.
```
Comment (nickwalton) [inline Source/Holodeck/ClientCommands/Private/RenderViewportCommand.cpp:19]: Did you mean to leave this in here?

**Review by nickwalton — CHANGES_REQUESTED** [review]
I just noticed a couple style things, but besides that it looks great! Good work!

File: Source/Holodeck/ClientCommands/Private/RenderViewportCommand.cpp (hunk 3 above)
Comment (jaydenmilne) [reply Source/Holodeck/ClientCommands/Private/RenderViewportCommand.cpp:19]: Yes, it's a warning for people who would want to copy this code

**Pull request merged**